packages/dart_frog_gen/lib/src/build_route_configuration.dart

```diff
@@ -73,47 +74,54 @@ List<RouteDirectory> _getRouteDirectories({
       ? directorySegment
       : '/$directorySegment';
   // Only add nested middleware -- global middleware is added separately.
-  MiddlewareFile? middleware;
+  MiddlewareFile? currentMiddleware;
   if (directory.path != path.join(Directory.current.path, 'routes')) {
     final middlewareFile = File(path.join(directory.path, '_middleware.dart'));
     if (middlewareFile.existsSync()) {
       final middlewarePath = path
           .relative(middlewareFile.path, from: routesDirectory.path)
           .replaceAll(r'\', '/');
-      middleware = MiddlewareFile(
+      currentMiddleware = MiddlewareFile(
         name: middlewarePath.toAlias(),
         path: path.join('..', 'routes', middlewarePath).replaceAll(r'\', '/'),
       );
-      onMiddleware(middleware);
+      onMiddleware(currentMiddleware);
     }
   }
 
+  final middleware = [
+    ...cascadingMiddleware,
+    if (currentMiddleware != null) currentMiddleware,
+  ];
+
   final files = _getRouteFiles(
     directory: directory,
     routesDirectory: routesDirectory,
     onRoute: onRoute,
     onRogueRoute: onRogueRoute,
   );
 
-  final baseRoute = directoryPath.toRoute();
-  for (final file in files) {
-    var endpoint = (baseRoute + file.route.toRoute()).replaceAll('//', '/');
-    if (endpoint.endsWith('/')) {
-      endpoint = endpoint.substring(0, endpoint.length - 1);
+  if (files.isNotEmpty) {
```

Comment:
optimization which allows us to avoid mounting routers for routes with no route handlers.